### ISSUE TITLE
Add a unique id to `Bet` class to make matching bets with langfuse traces easier

### ISSUE THREAD

**Issue by evangriffiths** [gnosis/prediction-market-agent-tooling#422]

**Comment by gabrielfior** [comment]
@evangriffiths correct me if I'm wrong, but I understood `transaction_hash` would be sufficient for uniquely identifying a bet?

**Comment by evangriffiths** [comment]
Yeah sure. There's already an ID associated with a bet from the subgraph (see https://github.com/gnosis/prediction-market-agent-tooling/blob/5fe7dfd40ecb9cf088c4a9c42d5f78fbfc3a3605/prediction_market_agent_tooling/markets/omen/data_models.py#L385). Hopefully this is the same as the transaction hash so that we can:

1. save the transaction hash to the `Trade` object that is 'observed' by langfuse
2. get the id from the `ResolvedBet`s from `OmenAgentMarket.get_resolved_bets_made_since`

and match these up. I shall check!
